### ISSUE TITLE
Docker fails to build when outside build context directory when using --stream

### ISSUE THREAD

**Issue by delissonjunio** [moby/moby#35242]
**Description**
Docker fails to build image when using --stream and setting a different parameter for build context location.

**Steps to reproduce the issue:**
1. `mkdir loc1; mkdir loc1/env`
2. `echo -e 'FROM ubuntu\nRUN sleep 1' >loc1/env/Dockerfile`
3. `cd loc1/; docker build env; echo 'stream: '; docker build --stream env`

**Describe the results you received:**
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM ubuntu
 ---> 747cb2d60bbe
Step 2/2 : RUN sleep 1
 ---> Using cache
 ---> 4cfe7feea9fb
Successfully built 4cfe7feea9fb
stream: 
failed to open Dockerfile: open Dockerfile: no such file or directory

**Describe the results you expected:**
Docker should find the Dockerfile in the specified build context directory, as happens when not using --stream flag.

**Additional information you deem important (e.g. issue happens only occasionally):**
If you create a symbolic link called Dockerfile in loc1 and run build with --stream, it works

**Output of `docker version`:**

```
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:18 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:56 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 3
 Running: 0
 Paused: 0
 Stopped: 3
Images: 12
Server Version: 17.09.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-96-generic
Operating System: elementary OS 0.4.1 Loki
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.54GiB
Name: pc-concertbh-delisson
ID: CQP4:APCT:RY4B:AVDG:BDUK:E54F:JWWS:OJNI:FW3B:T73Y:5UVX:OP4H
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running on physical host, also happens under virtualized environments

**Comment by jonastei** [comment]
Can confirm, happening here too

**Comment by delissonjunio** [comment]
Update: on docker server 17.10, this part doesn't error out but something else does:

Error response from daemon: failed to copy to /var/lib/docker/builder/f1580e4433cf21114883cbbdf18812c42fb0b298f3c7389f31548c2e652af418: rpc error: code = DeadlineExceeded desc = context deadline exceeded

While the client is "Streaming build context to Docker daemon". I've tried lowering the size of the data but it doesn't seem to matter (lowest I tried was a few hundred MB).

**Comment by thaJeztah** [comment]
ping @tonistiigi PTAL

**Comment by tonistiigi** [comment]
Added a commit with the fix to docker/cli#300

@delissonjunio was this also related to being outside of context dir? I still get the same error on `17.10` and building from context directory seems to work fine. So this may be something different. If so open a new issue with all the logs and version info and ping me there.

**Comment by delissonjunio** [comment]
@tonistiigi Created a [new issue](https://github.com/moby/moby/issues/35391), it seems to work on 17.10 so I'm closing this one.
